src/plugins/workspace/public/components/workspace_form/data_source_connection_table.tsx

```diff
@@ -3,106 +3,61 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import './data_source_connection_table.scss';
-import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import React, { useState } from 'react';
 import {
-  EuiSpacer,
   EuiInMemoryTable,
   EuiBasicTableColumn,
-  EuiTableSelectionType,
   EuiTableActionsColumnType,
-  EuiConfirmModal,
-  EuiSearchBarProps,
   EuiText,
   EuiListGroup,
   EuiListGroupItem,
   EuiPopover,
   EuiButtonEmpty,
   EuiPopoverTitle,
-  EuiSmallButton,
   EuiLink,
   EuiButtonIcon,
+  EuiInMemoryTableProps,
+  EuiTableSelectionType,
 } from '@elastic/eui';
 import { i18n } from '@osd/i18n';
 import { DataSourceConnection, DataSourceConnectionType } from '../../../common/types';
 import { AssociationDataSourceModalMode } from '../../../common/constants';
-import { DirectQueryConnectionIcon } from '../workspace_form';
+import { DirectQueryConnectionIcon } from './direct_query_connection_icon';
+import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
+import { CoreStart } from '../../../../../core/public';
 
-interface DataSourceConnectionTableProps {
+import './data_source_connection_table.scss';
+
+type DataSourceConnectionTableProps = Omit<
+  EuiInMemoryTableProps<DataSourceConnection>,
+  | 'columns'
+  | 'itemsId'
+  | 'isSelectable'
+  | 'itemIdToExpandedRowMap'
+  | 'isExpandable'
+  | 'selection'
+  | 'pagination'
+> & {
```

Comment:
```suggestion
type DataSourceConnectionTableProps = {
  tableProps: Partial<EuiInMemoryTableProps<DataSourceConnection>>
```
What about we moving all the table related props to a standalone entry so that it won't bother other props. And to make the props simple by using Partial.